src/main/java/seedu/address/model/client/policy/CustomDate.java

```diff
@@ -39,8 +41,17 @@ public CustomDate(String stringDate) {
      * @return LocalDate object.
      */
     public static LocalDate stringToDate(String date) {
-        DateTimeFormatter sf = DateTimeFormatter.ofPattern("dd.MM.yyyy");
-        LocalDate localDate = LocalDate.from(sf.parse(date));
+        LocalDate localDate;
```

Comment:
I don't think the checking of the leap year date should be here, it should be under `isValidDate()` if not many errors might occur over here because the data passed in this method should be valid already